api/rosetta/data_integration_test.go

```diff
@@ -97,6 +111,59 @@ func setupAPI(t *testing.T, db *badger.DB) *rosetta.Data {
 	return controller
 }
 
+func setupRecorder(endpoint string, input interface{}, options ...func(*http.Request)) (*httptest.ResponseRecorder, echo.Context, error) {
+
+	payload, ok := input.([]byte)
+	if !ok {
+		var err error
+		payload, err = json.Marshal(input)
+		if err != nil {
+			return nil, echo.New().AcquireContext(), fmt.Errorf("could not encode input: %w", err)
+		}
+	}
+
+	req := httptest.NewRequest(http.MethodPost, endpoint, bytes.NewReader(payload))
+	req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+	for _, opt := range options {
+		opt(req)
+	}
+
+	rec := httptest.NewRecorder()
+
+	ctx := echo.New().NewContext(req, rec)
+
+	return rec, ctx, nil
+}
+
+func checkRosettaError(statusCode int, def meta.ErrorDefinition) func(t assert.TestingT, err error, v ...interface{}) bool {
+
+	return func(t assert.TestingT, err error, v ...interface{}) bool {
+
+		// return false if any of the asserts failed
+
+		success := true
+
+		success = success && assert.Error(t, err)
+
+		echoErr, ok := err.(*echo.HTTPError)
+		if !assert.True(t, ok) {
+			return false
+		}
+
+		success = success && assert.Equal(t, statusCode, echoErr.Code)
+
+		gotErr, ok := echoErr.Message.(rosetta.Error)
+		if !assert.True(t, ok) {
+			return false
+		}
```

Comment:
For this kind of stuff we could also do

```go
if !assert.IsType(t, rosetta.Error{}, echoErr.Message) {
	return false
}
gotErr := echoErr.Message.(rosetta.Error)
```

Do you think it improves clarity? I'm not sure it's really useful in this case, but just putting it out there in case it's useful 📦

Reply:
Didn't know that one, makes sense.